tools/dump_fastssim.c

```diff
@@ -121,10 +121,14 @@ static void fs_downsample_level(fs_ctx *_ctx,int _l){
   int                 j;
   w=_ctx->level[_l].w;
   h=_ctx->level[_l].h;
+  w = (w >> 1) << 1;
+  h = (h >> 1) << 1;
```

Comment:
Why do you use bit shifting and not bitwise and operation here?  I would expect this operation to give the same result and be simpler to read:

w &= ~1